Jenkinsfile.migration

```diff
@@ -11,65 +11,120 @@
+							env.SRC_IS_LATEST = release.latestVersion
+							env.SRC_VERSION =  release.version
+							env.SRC_RELEASE =  release.nonVersionedFile
+						}
+
+					}
 				}
-				sh """
-				json_files_copy() {
-					case \$1 in
-						0* | master)
-							cp test/config.json .
-							cp test/genesisBlock.json .
-							;;
-						*)
-							cp test/data/config.json .
-							cp test/data/genesis_block.json .
-							;;
-					esac
+				stage('Create destination release') {
+					steps {
+						script {
+							def release = liskBuild(branch: params.MIGRATE_TO, network: 'test', useTestConfig: true)
+							env.DEST_IS_LATEST = release.latestVersion
+							env.DEST_VERSION =  release.version
+							env.DEST_RELEASE =  release.nonVersionedFile
+						}
+					}
 				}
+			}
+		}
 
-				dropdb lisk_test || true
-				createdb lisk_test
-				cd "${params.MIGRATE_FROM}"
-				json_files_copy "${params.MIGRATE_FROM}"
-				npm install
-				cd "../${params.MIGRATE_TO}"
-				json_files_copy "${params.MIGRATE_TO}"
-				npm install
-				sudo service postgresql restart
-				"""
+		stage("Install source release") {
+			steps {
+				installLisk(env.SRC_RELEASE, 'test')
 			}
 		}
-		stage("Start old Lisk") {
+
+		stage("Migrate to destination release") {
 			steps {
-				sh """
-				cd "${params.MIGRATE_FROM}"
-				pm2 start --name lisk app.js
-				sleep 20
-				pm2 kill
-				"""
+				script {
+					env.TARGET_BLOCK_HEIGHT = getBlockHeight() + 5
+				}
+				dir("lisk-home") {
+					sh """
+					cp ${env.DEST_RELEASE} .
+					wget https://raw.githubusercontent.com/LiskHQ/lisk-scripts/0.2.0/downloaded/lisk_bridge.sh
+					bash lisk_bridge.sh -f `basename ${env.DEST_RELEASE}` -n test -h ${env.TARGET_BLOCK_HEIGHT} -s ${WORKSPACE}/lisk-home/lisk-test
+					"""
+				}
 			}
 		}
-		stage("Start new Lisk") {
+
+		stage("Verify migration") {
 			steps {
+				script {
+					waitForBlockHeight(env.TARGET_BLOCK_HEIGHT.toInteger() + 5)
+				}
+			}
+		}
+	}
+	post {
+		always {
+			dir("lisk-home/lisk-test") {
 				sh """
-				cd "${params.MIGRATE_TO}"
-				pm2 start --name lisk app.js
-				sleep 30
-				if grep "error" logs/lisk.log; then
-					exit 1;
-				fi
+				bash lisk.sh stop
 				"""
 			}
+			cleanWs()
 		}
 	}
 }
+
+def installLisk(buildFile, network) {
+	dir("lisk-home") {
+		sh """
+		cp ${buildFile} .
+		wget https://raw.githubusercontent.com/LiskHQ/lisk-scripts/0.2.0/downloaded/installLisk.sh
+		bash installLisk.sh install -r ${network} -f `basename ${buildFile}`
+		"""
+	}
+}
+
+def getBlockHeight() {
+	dir("lisk-home/lisk-test") {
+		sh """
+		bash lisk.sh status | tail -1 | awk '{print \$4}' > .lisk-block-height
+		"""
+	}
+	return readFile("lisk-home/lisk-test/.lisk-block-height").trim().toInteger()
+}
+
+def waitForBlockHeight(targetBlockHeight, waitAttempts=0, previousHeight=0) {
+	def height = getBlockHeight()
+	echo "Height: ${height}"
+
+	if(previousHeight != height) {
+		previousHeight = height
+		waitAttempts = 0
+	} else if (waitAttempts < 3) {
```

Comment:
I can expect this part might be really fragile. 

The safer solution would be to check the height reached the target. Throw the error if it's already too high, but ignore potential downtimes which might occur.

Reply:
I can't find any way for `timeout` in Jenkins, so added this alternative. If blockchain is stuck and not moving forward, then only check for target height will never pass. So this code will check every 10s for 3 times if the block chain is moving forward, and if not then it will fail the job. I don't see its fragile.